## 收集数据
- tweet无法访问 :)
- 从github仓库[https://github.com/udacity/new-dand-advanced-china](https://github.com/udacity/new-dand-advanced-china)中下载文件

In [2]:
# -*- coding=utf-8 -*-
import requests

def download(url):
    with open(url.split('/')[-1], mode="wb") as f:
        respone = requests.get(url)
        f.write(respone.content)
        
# 下载 image-predictions.tsv
image_prediction_url = "https://raw.githubusercontent.com/udacity/new-dand-advanced-china/master/%E6%95%B0%E6%8D%AE%E6%B8%85%E6%B4%97/WeRateDogs%E9%A1%B9%E7%9B%AE/image-predictions.tsv"
download(image_prediction_url)
# 下载 twitter_archive_enhanced.csv
twitter_archive_enhanced_url = "https://raw.githubusercontent.com/udacity/new-dand-advanced-china/master/%E6%95%B0%E6%8D%AE%E6%B8%85%E6%B4%97/WeRateDogs%E9%A1%B9%E7%9B%AE/twitter-archive-enhanced.csv"
download(twitter_archive_enhanced_url)
# 下载 tweet_json.txt
tweet_json_url = "https://raw.githubusercontent.com/udacity/new-dand-advanced-china/master/%E6%95%B0%E6%8D%AE%E6%B8%85%E6%B4%97/WeRateDogs%E9%A1%B9%E7%9B%AE/tweet_json.txt"
download(tweet_json_url)

## 评估数据

In [3]:
# 使用pandas读取数据文件
import pandas as pd

In [4]:
image_predictions = pd.read_csv("image-predictions.tsv", sep='\t')
image_predictions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2075 entries, 0 to 2074
Data columns (total 12 columns):
tweet_id    2075 non-null int64
jpg_url     2075 non-null object
img_num     2075 non-null int64
p1          2075 non-null object
p1_conf     2075 non-null float64
p1_dog      2075 non-null bool
p2          2075 non-null object
p2_conf     2075 non-null float64
p2_dog      2075 non-null bool
p3          2075 non-null object
p3_conf     2075 non-null float64
p3_dog      2075 non-null bool
dtypes: bool(3), float64(3), int64(2), object(4)
memory usage: 152.0+ KB


In [5]:
image_predictions.head(3)

tweet_id                                          jpg_url  \
0  666020888022790149  https://pbs.twimg.com/media/CT4udn0WwAA0aMy.jpg   
1  666029285002620928  https://pbs.twimg.com/media/CT42GRgUYAA5iDo.jpg   
2  666033412701032449  https://pbs.twimg.com/media/CT4521TWwAEvMyu.jpg   

   img_num                      p1   p1_conf p1_dog                  p2  \
0        1  Welsh_springer_spaniel  0.465074   True              collie   
1        1                 redbone  0.506826   True  miniature_pinscher   
2        1         German_shepherd  0.596461   True            malinois   

    p2_conf p2_dog                   p3   p3_conf p3_dog  
0  0.156665   True    Shetland_sheepdog  0.061428   True  
1  0.074192   True  Rhodesian_ridgeback  0.072010   True  
2  0.138584   True           bloodhound  0.116197   True

In [8]:
# 处理和评估tweet_json.txt数据
import json
with open("tweet_json.txt", 'r') as json_file:
    for data in json_file.readlines():
        print(json.dumps(json.loads(data), indent=4, sort_keys=False, ensure_ascii=True))
        break

{
    "contributors": null, 
    "truncated": false, 
    "is_quote_status": false, 
    "in_reply_to_status_id": null, 
    "id": 892420643555336193, 
    "favorite_count": 39492, 
    "full_text": "This is Phineas. He's a mystical boy. Only ever appears in the hole of a donut. 13/10 https://t.co/MgUWQ76dJU", 
    "source": "<a href=\"http://twitter.com/download/iphone\" rel=\"nofollow\">Twitter for iPhone</a>", 
    "retweeted": false, 
    "coordinates": null, 
    "entities": {
        "symbols": [], 
        "user_mentions": [], 
        "hashtags": [], 
        "urls": [], 
        "media": [
            {
                "expanded_url": "https://twitter.com/dog_rates/status/892420643555336193/photo/1", 
                "display_url": "pic.twitter.com/MgUWQ76dJU", 
                "url": "https://t.co/MgUWQ76dJU", 
                "media_url_https": "https://pbs.twimg.com/media/DGKD1-bXoAAIAUK.jpg", 
                "id_str": "892420639486877696", 
                "sizes": {
    

In [40]:
# 逐行读取tweet_json.txt文件并添加到pandas DataFrame中，（至少）包含 tweet ID、retweet_count和favorite_count字段...
# 提出问题，放到list中，然后通过 'id_str': list1, 'fav_count': list2的方式创建df的效率是否更高？
df_tweet = pd.DataFrame()

# 观察tweet_json.txt后，对以下数据感兴趣：
# retweet_count favorite_count full_text retweeted source favorited
with open("tweet_json.txt", 'r') as json_file:
    index = ['id', 'retweet_count', 'favorite_count', 'full_text', 'retweeted', 'source', 'favorited']
    for data in json_file.readlines():
        dict_data = json.loads(data)
        
        data = []
        for idx in index:
            data.append(dict_data[idx])
        
        s_tweet = pd.Series(data, index=index)
        retweet_count = dict_data[u'retweet_count']
        df_tweet = df_tweet.append(s_tweet, ignore_index=True)

df_tweet.head(3)

favorite_count  favorited  \
0         39492.0        0.0   
1         33786.0        0.0   
2         25445.0        0.0   

                                           full_text            id  \
0  This is Phineas. He's a mystical boy. Only eve...  8.924206e+17   
1  This is Tilly. She's just checking pup on you....  8.921774e+17   
2  This is Archie. He is a rare Norwegian Pouncin...  8.918152e+17   

   retweet_count  retweeted                                             source  
0         8842.0        0.0  <a href="http://twitter.com/download/iphone" r...  
1         6480.0        0.0  <a href="http://twitter.com/download/iphone" r...  
2         4301.0        0.0  <a href="http://twitter.com/download/iphone" r...

In [41]:
df_tweet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2352 entries, 0 to 2351
Data columns (total 7 columns):
favorite_count    2352 non-null float64
favorited         2352 non-null float64
full_text         2352 non-null object
id                2352 non-null float64
retweet_count     2352 non-null float64
retweeted         2352 non-null float64
source            2352 non-null object
dtypes: float64(5), object(2)
memory usage: 128.7+ KB


In [42]:
df_tweet.describe()

favorite_count  favorited            id  retweet_count  retweeted
count     2352.000000     2352.0  2.352000e+03    2352.000000     2352.0
mean      8109.198980        0.0  7.425913e+17    3134.932398        0.0
std      11980.795669        0.0  6.846210e+16    5237.846296        0.0
min          0.000000        0.0  6.660209e+17       0.000000        0.0
25%       1417.000000        0.0  6.783949e+17     618.000000        0.0
50%       3596.500000        0.0  7.193536e+17    1456.500000        0.0
75%      10118.000000        0.0  7.991219e+17    3628.750000        0.0
max     132318.000000        0.0  8.924206e+17   79116.000000        0.0

In [28]:
df_tweet

favorite_count                                          full_text  \
0            39492.0  This is Phineas. He's a mystical boy. Only eve...   
1            33786.0  This is Tilly. She's just checking pup on you....   
2            25445.0  This is Archie. He is a rare Norwegian Pouncin...   
3            42863.0  This is Darla. She commenced a snooze mid meal...   
4            41016.0  This is Franklin. He would like you to stop ca...   
5            20548.0  Here we have a majestic great white breaching ...   
6            12053.0  Meet Jax. He enjoys ice cream so much he gets ...   
7            66596.0  When you watch your owner call another dog a g...   
8            28187.0  This is Zoey. She doesn't want to be one of th...   
9            32467.0  This is Cassie. She is a college pup. Studying...   
10           31127.0  This is Koda. He is a South Australian decksha...   
11           28208.0  This is Bruno. He is a service shark. Only get...   
12           38745.0  Here's a puppo that seems to be on the fence a...   
13           27633.0  This is Ted. He does his best. Sometimes that'...   
14           15329.0  This is Stuart. He's sporting his favorite fan...   
15           25712.0  This is Oliver. You're witnessing one of his m...   
16           29555.0  This is Jim. He found a fren. Taught him how t...   
17           26021.0  This is Zeke. He has a new stick. Very proud o...   
18           20267.0  This is Ralphus. He's powering up. Attempting ...   
19           22144.0  This is Gerald. He was just told he didn't get...   
20           30690.0  This is Jeffrey. He has a monopoly on the pool...   
21           46940.0  I've yet to rate a Venezuelan Hover Wiener. Th...   
22           70007.0  This is Canela. She attempted some fancy porch...   
23           34223.0  You may not have known you needed to see this ...   
24           31045.0  This... is a Jubilant Antarctic House Bear. We...   
25           35786.0  This is Maya. She's very shy. Rarely leaves he...   
26           12286.0  This is Mingus. He's a wonderful father to his...   
27           22802.0  This is Derek. He's late for a dog meeting. 13...   
28           21488.0  This is Roscoe. Another pupper fallen victim t...   
29             117.0  @NonWhiteHat @MayhewMayhem omg hello tanner yo...   
...              ...                                                ...   
2322           457.0  This is quite the dog. Gets really excited whe...   
2323           113.0  This is a southern Vesuvius bumblegruff. Can d...   
2324           171.0  Oh goodness. A super rare northeast Qdoba kang...   
2325           194.0  Those are sunglasses and a jean jacket. 11/10 ...   
2326           801.0  Unique dog here. Very small. Lives in containe...   
2327           228.0  Here we have a mixed Asiago from the Galápagos...   
2328           308.0  Look at this jokester thinking seat belt laws ...   
2329           203.0  This is an extremely rare horned Parthenon. No...   
2330           519.0  This is a funny dog. Weird toes. Won't come do...   
2331           152.0  This is an Albanian 3 1/2 legged  Episcopalian...   
2332           183.0     Can take selfies 11/10 https://t.co/ws2AMaNwPW   
2333           108.0  Very concerned about fellow dog trapped in com...   
2334         14703.0  Not familiar with this breed. No tail (weird)....   
2335            81.0  Oh my. Here you are seeing an Adobe Setter giv...   
2336           160.0  Can stand on stump for what seems like a while...   
2337           168.0  This appears to be a Mongolian Presbyterian mi...   
2338           121.0  Here we have a well-established sunblockerspan...   
2339           334.0  Let's hope this flight isn't Malaysian (lol). ...   
2340           154.0  Here we have a northern speckled Rhododendron....   
2341           494.0  This is the happiest dog you will ever see. Ve...   
2342           117.0  Here is the Rand Paul of retrievers folks! He'...   
2343           304.0  My oh my.

In [44]:
witter_archive_enhanced = pd.read_csv("twitter-archive-enhanced.csv")
witter_archive_enhanced.info()
witter_archive_enhanced.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2356 entries, 0 to 2355
Data columns (total 17 columns):
tweet_id                      2356 non-null int64
in_reply_to_status_id         78 non-null float64
in_reply_to_user_id           78 non-null float64
timestamp                     2356 non-null object
source                        2356 non-null object
text                          2356 non-null object
retweeted_status_id           181 non-null float64
retweeted_status_user_id      181 non-null float64
retweeted_status_timestamp    181 non-null object
expanded_urls                 2297 non-null object
rating_numerator              2356 non-null int64
rating_denominator            2356 non-null int64
name                          2356 non-null object
doggo                         2356 non-null object
floofer                       2356 non-null object
pupper                        2356 non-null object
puppo                         2356 non-null object
dtypes: float64(4), int64(3), ob

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
count  2.356000e+03           7.800000e+01         7.800000e+01   
mean   7.427716e+17           7.455079e+17         2.014171e+16   
std    6.856705e+16           7.582492e+16         1.252797e+17   
min    6.660209e+17           6.658147e+17         1.185634e+07   
25%    6.783989e+17                    NaN                  NaN   
50%    7.196279e+17                    NaN                  NaN   
75%    7.993373e+17                    NaN                  NaN   
max    8.924206e+17           8.862664e+17         8.405479e+17   

       retweeted_status_id  retweeted_status_user_id  rating_numerator  \
count         1.810000e+02              1.810000e+02       2356.000000   
mean          7.720400e+17              1.241698e+16         13.126486   
std           6.236928e+16              9.599254e+16         45.876648   
min           6.661041e+17              7.832140e+05          0.000000   
25%                    NaN                       NaN         10.000000   
50%                    NaN                       NaN         11.000000   
75%                    NaN                       NaN         12.000000   
max           8.874740e+17              7.874618e+17       1776.000000   

       rating_denominator  
count         2356.000000  
mean            10.455433  
std              6.745237  
min              0.000000  
25%             10.000000  
50%             10.000000  
75%             10.000000  
max            170.000000

In [45]:
witter_archive_enhanced

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
5     891087950875897856                    NaN                  NaN   
6     890971913173991426                    NaN                  NaN   
7     890729181411237888                    NaN                  NaN   
8     890609185150312448                    NaN                  NaN   
9     890240255349198849                    NaN                  NaN   
10    890006608113172480                    NaN                  NaN   
11    889880896479866881                    NaN                  NaN   
12    889665388333682689                    NaN                  NaN   
13    889638837579907072                    NaN                  NaN   
14    889531135344209921                    NaN                  NaN   
15    889278841981685760                    NaN                  NaN   
16    888917238123831296                    NaN                  NaN   
17    888804989199671297                    NaN                  NaN   
18    888554962724278272                    NaN                  NaN   
19    888202515573088257                    NaN                  NaN   
20    888078434458587136                    NaN                  NaN   
21    887705289381826560                    NaN                  NaN   
22    887517139158093824                    NaN                  NaN   
23    887473957103951883                    NaN                  NaN   
24    887343217045368832                    NaN                  NaN   
25    887101392804085760                    NaN                  NaN   
26    886983233522544640                    NaN                  NaN   
27    886736880519319552                    NaN                  NaN   
28    886680336477933568                    NaN                  NaN   
29    886366144734445568                    NaN                  NaN   
...                  ...                    ...                  ...   
2326  666411507551481857                    NaN                  NaN   
2327  666407126856765440                    NaN                  NaN   
2328  666396247373291520                    NaN                  NaN   
2329  666373753744588802                    NaN                  NaN   
2330  666362758909284353                    NaN                  NaN   
2331  666353288456101888                    NaN                  NaN   
2332  666345417576210432                    NaN                  NaN   
2333  666337882303524864                    NaN                  NaN   
2334  666293911632134144                    NaN                  NaN   
2335  666287406224695296                    NaN                  NaN   
2336  666273097616637952                    NaN                  NaN   
2337  666268910803644416                    NaN                  NaN   
2338  666104133288665088                    NaN                  NaN   
2339  666102155909144576                    NaN                  NaN   
2340  666099513787052032                    NaN                  NaN   
2341  666094000022159362                    NaN                  NaN   
2342  666082916733198337                    NaN                  NaN   
2343  666073100786774016                    NaN                  NaN   
2344  666071193221509120                    NaN                  NaN   
2345  666063827256086533                    NaN                  NaN   
2346  666058600524156928                    NaN                  NaN   
2347  666057090499244032                    NaN                  NaN   
2348  666055525042405380                    NaN                  NaN   
2349  666051853826850816                    NaN         

#### 质量
##### `df_tweet` 表格
- favorite_count、id、retweet_count应该是int64类型，而不是浮点浮点类型
- favorited 和 retweeted 的值都是零，需要删除这两列

##### `twitter_archive_enhanced` 表格
- doggo floofer pupper puppo字段有"None"这是python关键字，在csv体现为字符串，这应该在表示空
- in_reply_to_status_id和in_reply_to_user_id，只有78个值，这两个列的数据很可能没用
- timestamp、retweeted_status_timestamp应该是时间类型，现在是object类型
- rating_denominator不是10的需要清理掉, 比如170是最大值是不对的
- rating_numerator分子小于rating_denominator的需要清理掉
- expanded_urls的单条记录中有重复的url地址，清理清理只保留一个

#### 清洁度
##### `df_tweet` 表格
- source字段内容是html文本，需要更加简洁，只需要获取该html的text部分即可
- 通过witter_archive_enhanced.tweet_id和df_tweet.id和image_predictions.tweet_id关联创建新的表格

## 清理

In [46]:
# 清理前的备份
df_tweet_clean = df_tweet.copy()
witter_archive_enhanced_clean = witter_archive_enhanced.copy()

In [47]:
# favorite_count id retweet_count应该是int64类型，现在做格式转换 long
for i in ['favorite_count', 'id', 'retweet_count']:
    df_tweet_clean[i] = df_tweet_clean[i].astype(long)
df_tweet_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2352 entries, 0 to 2351
Data columns (total 7 columns):
favorite_count    2352 non-null int64
favorited         2352 non-null float64
full_text         2352 non-null object
id                2352 non-null int64
retweet_count     2352 non-null int64
retweeted         2352 non-null float64
source            2352 non-null object
dtypes: float64(2), int64(3), object(2)
memory usage: 128.7+ KB


In [48]:
# 顺便调整列的顺序
index = ['id', 'retweet_count', 'favorite_count', 'full_text', 'source']
df_tweet_clean = df_tweet_clean.reindex_axis(index, axis=1)
df_tweet_clean

id  retweet_count  favorite_count  \
0     892420643555336192           8842           39492   
1     892177421306343424           6480           33786   
2     891815181378084864           4301           25445   
3     891689557279858688           8925           42863   
4     891327558926688256           9721           41016   
5     891087950875897856           3240           20548   
6     890971913173991424           2142           12053   
7     890729181411237888          19548           66596   
8     890609185150312448           4403           28187   
9     890240255349198848           7684           32467   
10    890006608113172480           7584           31127   
11    889880896479866880           5116           28208   
12    889665388333682688           8502           38745   
13    889638837579907072           4705           27633   
14    889531135344209920           2309           15329   
15    889278841981685760           5635           25712   
16    888917238123831296           4681           29555   
17    888804989199671296           4535           26021   
18    888554962724278272           3722           20267   
19    888078434458587136           3637           22144   
20    887705289381826560           5584           30690   
21    887517139158093824          12053           46940   
22    887473957103951872          18813           70007   
23    887343217045368832          10713           34223   
24    887101392804085760           6147           31045   
25    886983233522544640           8045           35786   
26    886736880519319552           3420           12286   
27    886680336477933568           4597           22802   
28    886366144734445568           3297           21488   
29    886267009285017600              4             117   
...                  ...            ...             ...   
2322  666411507551481856            337             457   
2323  666407126856765440             43             113   
2324  666396247373291520             91             171   
2325  666373753744588800             99             194   
2326  666362758909284352            590             801   
2327  666353288456101888             76             228   
2328  666345417576210432            146             308   
2329  666337882303524864             96             203   
2330  666293911632134144            365             519   
2331  666287406224695296             71             152   
2332  666273097616637952             81             183   
2333  666268910803644416             37             108   
2334  666104133288665088           6835           14703   
2335  666102155909144576             15              81   
2336  666099513787052032             73             160   
2337  666094000022159360             78             168   
2338  666082916733198336             47             121   
2339  666073100786774016            173             334   
2340  666071193221509120             67             154   
2341  666063827256086528            230             494   
2342  666058600524156928             61             117   
2343  666057090499244032            146             304   
2344  666055525042405376            261             449   
2345  666051853826850816            877            1250   
2346  666050758794694656             60             136   
2347  666049248165822464             41             111   
2348  666044226329800704            147             309   
2349  666033412701032448             47             128   
2350  666029285002620928             48             132   
2351  666020888022790144            530            2528   

                                              full_text  \
0     This is Phineas. He's a mystical boy. Only eve...   
1     This is Tilly. She's just checking pup on you....   
2     This is Archie. He is a rare Norwegian Pouncin...   
3     This is Darla. She commenced a snooze mid meal...   
4     This is Franklin. He would like you to stop ca...   
5     Her

In [49]:
# rating_denominator不是10的需要清理掉
witter_archive_enhanced_clean = witter_archive_enhanced[witter_archive_enhanced["rating_denominator"]==10]
witter_archive_enhanced_clean.info()
witter_archive_enhanced_clean

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2333 entries, 0 to 2355
Data columns (total 17 columns):
tweet_id                      2333 non-null int64
in_reply_to_status_id         73 non-null float64
in_reply_to_user_id           73 non-null float64
timestamp                     2333 non-null object
source                        2333 non-null object
text                          2333 non-null object
retweeted_status_id           180 non-null float64
retweeted_status_user_id      180 non-null float64
retweeted_status_timestamp    180 non-null object
expanded_urls                 2278 non-null object
rating_numerator              2333 non-null int64
rating_denominator            2333 non-null int64
name                          2333 non-null object
doggo                         2333 non-null object
floofer                       2333 non-null object
pupper                        2333 non-null object
puppo                         2333 non-null object
dtypes: float64(4), int64(3), ob

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
5     891087950875897856                    NaN                  NaN   
6     890971913173991426                    NaN                  NaN   
7     890729181411237888                    NaN                  NaN   
8     890609185150312448                    NaN                  NaN   
9     890240255349198849                    NaN                  NaN   
10    890006608113172480                    NaN                  NaN   
11    889880896479866881                    NaN                  NaN   
12    889665388333682689                    NaN                  NaN   
13    889638837579907072                    NaN                  NaN   
14    889531135344209921                    NaN                  NaN   
15    889278841981685760                    NaN                  NaN   
16    888917238123831296                    NaN                  NaN   
17    888804989199671297                    NaN                  NaN   
18    888554962724278272                    NaN                  NaN   
19    888202515573088257                    NaN                  NaN   
20    888078434458587136                    NaN                  NaN   
21    887705289381826560                    NaN                  NaN   
22    887517139158093824                    NaN                  NaN   
23    887473957103951883                    NaN                  NaN   
24    887343217045368832                    NaN                  NaN   
25    887101392804085760                    NaN                  NaN   
26    886983233522544640                    NaN                  NaN   
27    886736880519319552                    NaN                  NaN   
28    886680336477933568                    NaN                  NaN   
29    886366144734445568                    NaN                  NaN   
...                  ...                    ...                  ...   
2325  666418789513326592                    NaN                  NaN   
2326  666411507551481857                    NaN                  NaN   
2327  666407126856765440                    NaN                  NaN   
2328  666396247373291520                    NaN                  NaN   
2329  666373753744588802                    NaN                  NaN   
2330  666362758909284353                    NaN                  NaN   
2331  666353288456101888                    NaN                  NaN   
2332  666345417576210432                    NaN                  NaN   
2333  666337882303524864                    NaN                  NaN   
2334  666293911632134144                    NaN                  NaN   
2336  666273097616637952                    NaN                  NaN   
2337  666268910803644416                    NaN                  NaN   
2338  666104133288665088                    NaN                  NaN   
2339  666102155909144576                    NaN                  NaN   
2340  666099513787052032                    NaN                  NaN   
2341  666094000022159362                    NaN                  NaN   
2342  666082916733198337                    NaN                  NaN   
2343  666073100786774016                    NaN                  NaN   
2344  666071193221509120                    NaN                  NaN   
2345  666063827256086533                    NaN                  NaN   
2346  666058600524156928                    NaN                  NaN   
2347  666057090499244032                    NaN                  NaN   
2348  666055525042405380                    NaN                  NaN   
2349  666051853826850816                    NaN         

In [50]:
# rating_numerator分子小于rating_denominator的需要清理掉
witter_archive_enhanced_clean = witter_archive_enhanced_clean[witter_archive_enhanced_clean["rating_numerator"] > witter_archive_enhanced_clean["rating_denominator"]]
witter_archive_enhanced_clean.info()
witter_archive_enhanced_clean

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1438 entries, 0 to 2339
Data columns (total 17 columns):
tweet_id                      1438 non-null int64
in_reply_to_status_id         56 non-null float64
in_reply_to_user_id           56 non-null float64
timestamp                     1438 non-null object
source                        1438 non-null object
text                          1438 non-null object
retweeted_status_id           153 non-null float64
retweeted_status_user_id      153 non-null float64
retweeted_status_timestamp    153 non-null object
expanded_urls                 1398 non-null object
rating_numerator              1438 non-null int64
rating_denominator            1438 non-null int64
name                          1438 non-null object
doggo                         1438 non-null object
floofer                       1438 non-null object
pupper                        1438 non-null object
puppo                         1438 non-null object
dtypes: float64(4), int64(3), ob

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
5     891087950875897856                    NaN                  NaN   
6     890971913173991426                    NaN                  NaN   
7     890729181411237888                    NaN                  NaN   
8     890609185150312448                    NaN                  NaN   
9     890240255349198849                    NaN                  NaN   
10    890006608113172480                    NaN                  NaN   
11    889880896479866881                    NaN                  NaN   
12    889665388333682689                    NaN                  NaN   
13    889638837579907072                    NaN                  NaN   
14    889531135344209921                    NaN                  NaN   
15    889278841981685760                    NaN                  NaN   
16    888917238123831296                    NaN                  NaN   
17    888804989199671297                    NaN                  NaN   
18    888554962724278272                    NaN                  NaN   
19    888202515573088257                    NaN                  NaN   
20    888078434458587136                    NaN                  NaN   
21    887705289381826560                    NaN                  NaN   
22    887517139158093824                    NaN                  NaN   
23    887473957103951883                    NaN                  NaN   
24    887343217045368832                    NaN                  NaN   
25    887101392804085760                    NaN                  NaN   
26    886983233522544640                    NaN                  NaN   
27    886736880519319552                    NaN                  NaN   
28    886680336477933568                    NaN                  NaN   
29    886366144734445568                    NaN                  NaN   
...                  ...                    ...                  ...   
2201  668645506898350081                    NaN                  NaN   
2212  668587383441514497                    NaN                  NaN   
2213  668567822092664832                    NaN                  NaN   
2217  668528771708952576                    NaN                  NaN   
2221  668480044826800133                    NaN                  NaN   
2225  668286279830867968                    NaN                  NaN   
2228  668256321989451776                    NaN                  NaN   
2234  668190681446379520                    NaN                  NaN   
2244  667886921285246976                    NaN                  NaN   
2250  667832474953625600                    NaN                  NaN   
2252  667801013445750784                    NaN                  NaN   
2257  667728196545200128                    NaN                  NaN   
2259  667550904950915073                    NaN                  NaN   
2267  667524857454854144                    NaN                  NaN   
2269  667509364010450944                    NaN                  NaN   
2270  667502640335572993                    NaN                  NaN   
2273  667470559035432960                    NaN                  NaN   
2275  667453023279554560                    NaN                  NaN   
2278  667435689202614272                    NaN                  NaN   
2283  667200525029539841                    NaN                  NaN   
2284  667192066997374976                    NaN                  NaN   
2292  667160273090932737                    NaN                  NaN   
2293  667152164079423490                    NaN                  NaN   
2301  667044094246576128                    NaN         

In [56]:
# expanded_urls的单条记录中有重复的url地址，清理清理只保留一个
witter_archive_enhanced_clean['expanded_urls'] = witter_archive_enhanced_clean['expanded_urls'].str.split(',')[0]
witter_archive_enhanced_clean

ValueError: Length of values does not match length of index